sourcegraph/sourcegraph
# Aggregation: Integrate aggregation UI and GQL API

Closes https://github.com/sourcegraph/sourcegraph/issues/39956
Closes https://github.com/sourcegraph/sourcegraph/issues/39958
Part of https://github.com/sourcegraph/sourcegraph/issues/40766

## Backround
This PR connects GQL Aggregation API with aggregation UI on the frontend. I recorded a [loom](https://www.loom.com/share/8dc8de38cbc54d87bf57f180d7bd9e62) in order to describe changes that this PR makes and the current state of aggregation project. 

## Todo 
- [x] Connect GQL API to the aggregation UI 
- [x] Adjust UI based on the lates [Figma designs](https://www.figma.com/file/cKSeCtmBh9aiPsAVoKsLgM/Aggregation-insights%3A-display-%22grouped%22-insights-for-existing-searches-%2339126-%5Bready-for-dev%5D?node-id=1913%3A35438)
- [ ] Add Info icon UI (will be addressed in this issue https://github.com/sourcegraph/sourcegraph/issues/40766 later)
- [x] Fix storybook stories
- [x] Fix existing integration tests
- [ ] Add new integration tests for aggregation data fetching
  
## Test plan
- Make sure that FE handler all possible states in aggregation UI appropriate (as it descirbed in Figma designs)
- Make sure that repositories, file name and author aggregation types work correctly 
- Test that side panel and expanded (the full mode) view don't have any visual regressions 



## App preview:

- [Web](https://sg-web-vk-integrate-aggregation-api.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-nitniwbbqf.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
